src/fondant/pipeline/compiler.py

```diff
@@ -639,19 +639,24 @@ def _check_ecr_pull_through_rule(self) -> None:
     def _patch_uri(self, og_uri: str) -> str:
         full_ref, tag = og_uri.split(":")
 
-        ref, *_ = full_ref.split("/")
+        ref, *repo = full_ref.split("/")
 
-        if ref == "fndnt":
-            logging.info("Reusable component detected, patching URI")
-            # force pullthrough cache to be used
+        def pull_through(repository_name):
             _ = self.ecr_client.batch_get_image(
-                repositoryName=f"{self.ecr_namespace}/{full_ref}",
+                repositoryName=repository_name,
                 imageIds=[{"imageTag": tag}],
             )
             repo_response = self.ecr_client.describe_repositories(
-                repositoryNames=[f"{self.ecr_namespace}/{full_ref}"],
+                repositoryNames=[repository_name],
             )
-            uri = repo_response["repositories"][0]["repositoryUri"] + ":" + tag
+            return repo_response["repositories"][0]["repositoryUri"] + ":" + tag
+
+        if ref == "fndnt":
+            logging.info("Reusable component detected, patching URI")
+            uri = pull_through(f"{self.ecr_namespace}/{full_ref}")
+        elif ref == "public.ecr.aws":
+            logging.info("Public AWS ECR component detected, patching URI")
+            uri = pull_through(f"{self.ecr_namespace}/{'/'.join(repo)}")
         else:
             logging.info("Custom component detected")
```

Comment:
Not related to your changes, but shouldn't we test here that the image is in a AWS registry any way? If the custom image isn't stored in ECR it will not work, right?

Reply:
Typically those things checks occur at runtime (e.g. in KFP). The local service account available at compile time might differ from the one at runtime and they may have different access rights to the custom image.

Reply:
It is also the user responsibility to make sure the component is available here, we have to assume it is there. 
We have docs on this: https://github.com/ml6team/fondant/blob/main/docs/runners/sagemaker.md#using-custom-fondant-components-on-sagemaker